gpytorch/utils/cholesky.py

```diff
@@ -22,6 +24,12 @@ def psd_safe_cholesky(A, upper=False, out=None, jitter=None):
         L = torch.cholesky(A, upper=upper, out=out)
         return L
     except RuntimeError as e:
+        isnan = torch.isnan(A)
```

Comment:
As long as we are doing n^2 work to check this, maybe we should do `torch.isnan(A.sum(-1))` or something to avoid materializing n^2 memory too

Reply:
Approved the PR since it shouldn't be a big deal though -- people likely shouldn't be using Cholesky at the size where GPU memory is a concern. It's just technically possible to check for NaNs without duplicating memory.

Reply:
Yeah that's a fair point.

Reply:
FWIW I raised this issue with the pytorch folks hopefully we can get them to implement a better error message in the case of NaNs upstream.